### ISSUE TITLE
Some suggestions for the sand table folder

### ISSUE THREAD

**Issue by Taeyeonzpx** [sandboxie-plus/Sandboxie#2819]
### Is your feature request related to a problem or use case?

I have a question, why is the folder of the sand table always in the designated sand table folder?  I want him to implement that the files on drive c are in the sand folder on drive c and the files on drive D are in the sand folder on drive D for example
C:\DefaultBox\1.txt
D:\DefaultBox\2.txt

### Describe the solution you'd like

The realization of the same sand table file can be distributed in different C disk,D disk, which seems to help the speed of rapid recovery

**Comment by bastik-1001** [comment]
I am sorry, if I don't understand what you are asking for, but it appears as if that has been asked in #2709.

`FileRootPath=\??\%SystemDrive%\Sandbox\%USER%\%SANDBOX% `is the default location, but `FileRootPath=`  can be set per sandbox to point to other folders or drives, like `FileRootPath=D:\%SANDBOX%\` (This can be done manually in the Sandboxie.ini)

In the issue, I linked to, it can be seen that you are able to select the root folder, when you use the new box wizard, while creating a new box.

Files written in those sandboxes, will be written into the sandbox folder for each sandbox, even if they are stored on a different drive, to have all the resources in one location.

**Comment by Taeyeonzpx** [comment]
> I am sorry, if I don't understand what you are asking for, but it appears as if that has been asked in #2709.
> 
> `FileRootPath=\??\%SystemDrive%\Sandbox\%USER%\%SANDBOX% `is the default location, but `FileRootPath=` can be set per sandbox to point to other folders or drives, like `FileRootPath=D:\%SANDBOX%\` (This can be done manually in the Sandboxie.ini)
> 
> In the issue, I linked to, it can be seen that you are able to select the root folder, when you use the new box wizard, while creating a new box.
> 
> Files written in those sandboxes, will be written into the sandbox folder for each sandbox, even if they are stored on a different drive, to have all the resources in one location.

No, that's not what I mean.  What I mean is the opposite of specifying the Sandbox root.  I'm currently running a program in the sandbox, and the program that is running in the sandbox is currently releasing files in the sandbox root ("E:\Sandbox\").  For example, "E:\Sandbox\DefaultBox\drive\C" change to
"C:\Sandbox\DefaultBox\drive" or "E:\Sandbox\DefaultBox\drive\D" change to
D:\Sandbox\DefaultBox\drive
I mean that the current sandbox root folder," drive "directory "C","D","E" drive letters, respectively into the corresponding disk sandbox folder
This won't cause my sandbox to occupy too much space on the same drive.  If so, the same drive copy and quick recovery seems to improve speed.  Right?

**Comment by Taeyeonzpx** [comment]
> I am sorry, if I don't understand what you are asking for, but it appears as if that has been asked in #2709.
> 
> `FileRootPath=\??\%SystemDrive%\Sandbox\%USER%\%SANDBOX% `is the default location, but `FileRootPath=` can be set per sandbox to point to other folders or drives, like `FileRootPath=D:\%SANDBOX%\` (This can be done manually in the Sandboxie.ini)
> 
> In the issue, I linked to, it can be seen that you are able to select the root folder, when you use the new box wizard, while creating a new box.
> 
> Files written in those sandboxes, will be written into the sandbox folder for each sandbox, even if they are stored on a different drive, to have all the resources in one location.

No, that's not what I mean. What I mean is the opposite of specifying the Sandbox root directory. Currently I am running a program in the sandbox, and the program that is running in the sandbox releases files that are currently in the sandbox root directory ("E:\Sandbox\"). For example, "E:\Sandbox\DefaultBox\drive\C" change to
"C:\Sandbox\DefaultBox\drive" or "E:\Sandbox\DefaultBox\drive\D" change to
D:\Sandbox\DefaultBox\drive
I mean that the current sandbox root folder," drive "directory "C","D","E" drive letters, respectively into the corresponding disk sandbox folder
This won't cause my sandbox to occupy too much space on the same drive. If so, the same drive copy and quick recovery seems to improve speed. Right?

**Comment by bastik-1001** [comment]
I misunderstood what you have asked for, initially. Thank you for explaining it again and for giving another example.

Now I understand the use-case and see the potential of not only using one drive's space and it being quicker to recover files that are on the same drive.

**Comment by bastik-1001** [comment]
David will have to comment on the technical aspect of this. Maybe creating snapshots could be more complicated (although that should be easy since those are just different sources), or restoring them. It might also be an issue if some files are located on drives that are not always connected to the computer.

If that is something that works, I can see that being an option for users that desire Sandboxie to behave this way.

**Comment by DavidXanatos** [comment]
Implementing this feature would be doable although it would complicate a couple of things, also in the driver and all the access mechanics as we would have not one writable root but one per drive.
It surely makes the boxes less portable, although since we now have an import/export function its somewhat not that bad as it would be otherwise.

Wouldn't it be a simple solution to just putt the sandbox root on your biggest ssd and that's that, recovery spreads should be very fast either way with modern SSD's reaching > 1GB/s

I would say if there are enough propel requesting this feature I could add it, but for a single request I don't think the advantages justify the afford.

Now that I think about it, it would not play nice with removable media as then these would get a sandbox directory, etc... although that might be actually desirable als an alternative to the volume serial number feature we have now.

**Comment by Taeyeonzpx** [comment]
> Implementing this feature would be doable although it would complicate a couple of things, also in the driver and all the access mechanics as we would have not one writable root but one per drive. It surely makes the boxes less portable, although since we now have an import/export function its somewhat not that bad as it would be otherwise.
> 
> Wouldn't it be a simple solution to just putt the sandbox root on your biggest ssd and that's that, recovery spreads should be very fast either way with modern SSD's reaching > 1GB/s
> 
> I would say if there are enough propel requesting this feature I could add it, but for a single request I don't think the advantages justify the afford.
> 
> Now that I think about it, it would not play nice with removable media as then these would get a sandbox directory, etc... although that might be actually desirable als an alternative to the volume serial number feature we have now.

I have some ideas and solutions for this,
Idea: 
Portable files we can keep the original solution to solve the problem, the current original solution, fast recovery speed is fine for SSD, but for HDD such as poor performance devices, or USB flash disk, it may lead to too long recovery time, slow system and other problems, resulting in poor sandbox performance
Solution:
We can emulate VMware's solution by defining a root folder, where the vmdk file corresponds to the drive letter in the drive folder of the sandbox. When creating a sandbox, the default drive folder is in the root directory of the sandbox. Alternatively, we can add a specific drive folder function when creating a sandbox. For the sandbox we have already created, we can add a function similar to the one introduced, Corresponding to VMware's disk addition function, if this solution is adopted, it will retain the original solution, ensure the portability of the sandbox, and also solve the performance issues of mobile devices such as HDDs or USB drives
